### PR TITLE
Volumes API

### PR DESCRIPTION
**Pull request by strokine**
Adding support for Volumes API.

### CODE REVIEW

**Comment by jeevatkm** [comment]
@strokine nice (#54), thanks for PR. I will have a look and get back to you.

**Comment by strokine** [comment]
Committed code for Volume Actions as well.

**Comment by strokine** [comment]
Completed all Block Storage API.

File: src/main/java/com/myjeeva/digitalocean/impl/DigitalOceanClient.java [hunk 1]
```diff
@@ -1056,8 +1081,130 @@ public Response untagResources(String name, List<Resource> resources)
...
+		  
+		Map<String, String> qp = new HashMap<String, String>();
+		qp.put("region", regionSlug);
+		return (Volumes) perform(new ApiRequest(ApiAction.AVAILABLE_VOLUMES, qp)).getData();
+	}
+	
+	@Override
+	public Volume createVolume(Volume volume) throws DigitalOceanException, RequestUnsuccessfulException {
+	    if (null == volume
+	        || StringUtils.isEmpty(volume.getName())
```
Comment (jeevatkm) [inline src/main/java/com/myjeeva/digitalocean/impl/DigitalOceanClient.java]: Please change it to `StringUtils.isBlank()`.

File: src/main/java/com/myjeeva/digitalocean/pojo/Volume.java [hunk 2]
```diff
@@ -0,0 +1,85 @@
...
+	
+	@SerializedName("droplet_ids")
+	private Set<Integer> dropletIds;
+	
+	private String name;
+	
+	private String description;
+	
+	@SerializedName("size_gigabytes")
+	private Integer sizeGigabytes;
```
Comment (jeevatkm) [inline src/main/java/com/myjeeva/digitalocean/pojo/Volume.java]: Please change datatype and variable name like [here](https://github.com/jeevatkm/digitalocean-api-java/blob/master/src/main/java/com/myjeeva/digitalocean/pojo/Image.java#L64). Thanks.

File: src/main/java/com/myjeeva/digitalocean/pojo/Volume.java [hunk 3]
```diff
@@ -0,0 +1,85 @@
...
+	
+	@SerializedName("created_at")
+	private Date createdAt;
```
Comment (jeevatkm) [inline src/main/java/com/myjeeva/digitalocean/pojo/Volume.java]: Currently [this](https://github.com/jeevatkm/digitalocean-api-java/blob/master/src/main/java/com/myjeeva/digitalocean/pojo/Image.java#L58) name is followed for `created_at` attribute in all POJOs.

It's better to keep consistence.  

_PS: IMHO current variable name is nothing wrong_

File: src/main/java/com/myjeeva/digitalocean/pojo/VolumeAction.java [hunk 4]
```diff
@@ -0,0 +1,93 @@
...
+	@SerializedName("region")
+	private String regionSlug;
+	
+	@Expose
+	@SerializedName("volume_name")
+	private String volumeName;
+	
+	@Expose
+	@SerializedName("size_gigabytes")
+	private Integer sizeGigabytes;
```
Comment (jeevatkm) [inline src/main/java/com/myjeeva/digitalocean/pojo/VolumeAction.java]: Please change datatype and variable name like [here](https://github.com/jeevatkm/digitalocean-api-java/blob/master/src/main/java/com/myjeeva/digitalocean/pojo/Image.java#L64). Thanks.

**Comment by jeevatkm** [comment]
@strokine Thanks for complete implementation.  I have added a notes, please take care.

Also, I couldn't see any integration tests, could you please add it [here](https://github.com/jeevatkm/digitalocean-api-java/blob/master/src/test/java/com/myjeeva/digitalocean/DigitalOceanIntegrationTest.java)?

**Comment by jeevatkm** [comment]
I forget to mention. Please apply Google Java Code Formatter, XML link is at [contributing section](https://github.com/jeevatkm/digitalocean-api-java#contributing). 

_PS: While applying code formatting in `enum` class, kindly select non-enum code lines and apply it otherwise it will spoil the enum alignments._

**Comment by strokine** [comment]
Hey @jeevatkm , I've refactored the code as you suggested and added integration tests.
Before you run the IntTests you have to create 2 volumes, and provide ID from one, and Name from another. This way the IntTests should run better.

RE: Google Java Code Formatter, I don't have it set up on my machine, maybe you could just apply it yourself, or as an option you could use this plugin here? https://github.com/revelc/formatter-maven-plugin

Thanks for the library.
Eugene

**Comment by jeevatkm** [comment]
@strokine Thanks again for PR.

I will take care -
* Code formatting
* `sizeGigabytes` -> `size`
* Javadoc

**Pull request merged**